lib/installHelpers.js

```diff
@@ -578,3 +580,89 @@ function error(msg) {
 function isSilent() {
   return process.env.SILENT;
 }
+
+/**
+* Manually collect all missing prerequisites
+* Therefore we do not pass error to async-callbacks as this would exit the process
+*/
+function checkDependencies(checks, message, callback) {
+  const result = [];
+  async.applyEach(checks, result, function(error) {
+    if (error) return callback(error);
+    if (result.length) {
+      result.unshift(message);
+      return callback(result.join('\n'));
+    }
+    return callback(null);
+  });
+}
+
+function checkRequiredDependencies(callback) {
+  checkDependencies([
+    checkGitInstalled,
+    checkGruntCliInstalled,
+    checkGithubConnection
+  ],
+  'Missing the following pre-requisites, please check install guides:',
+  callback);
+}
+function checkOptionalDependencies(callback) {
+  checkDependencies([
+    checkLocalMongoDbConnection,
+    checkFfmpegInstalled
+  ],
+  'Missing the following optional dependecies, please check install guides:',
+  callback);
+}
+
+function checkGitInstalled(result, gitCb) {
+  execCommand('git --version', {}, function(error, stdOut) {
+    if (error) {
+      result.push('- git');
+    }
+    gitCb();
+  });
+}
+
+function checkGruntCliInstalled(result, gruntCb) {
+  execCommand('grunt --version', {}, function(error, stdOut) {
+    if (error) {
+      result.push('- grunt-cli');
+    }
+    gruntCb();
+  });
+}
+
+function checkGithubConnection(result, githubCb) {
+  const url = 'https://api.github.com/';
+  request({
+    headers: {
+      'User-Agent': DEFAULT_USER_AGENT
+    },
+    uri: url,
+    method: 'GET'
+  }, function(error, response) {
+    if (error) {
+      result.push(`- failed to connect to ${url}`);
+    }
+    githubCb();
+  });
+}
+
+function checkLocalMongoDbConnection(result, callback) {
```

Comment:
This doesn't actually check the connection does it? Maybe needs a rename to avoid any confusion.